File: List/pythonic/TNL.py
from LinkedList import LinkedList
from ArrayList import ArrayList

def main():
    k = int(input())
    l = int(input())
    #bs = LinkedList()
    bs = ArrayList()
    for value in list(map(int, input().strip().split()[:l])):
        bs.append(value)
    m = int(input())
    #ts = LinkedList()
    ts = LinkedList()
    for value in list(map(int, input().strip().split()[:m])):
        ts.append(value)
    task = int(input())
    if task == 1:
        for i in range(k):
            print(i, end=('\n' if i == k-1 else ','))
        for i in range(k):
            temp = ',' if bs.Search(i) == -1 else i
            print(temp,end='')
            if temp != ',':
                print(end=('\n' if i == k-1 else ','))
        for i in range(k):
            temp = ',' if ts.Search(i) == -1 else i
            print(temp,end='')
            if temp != ',':
                print(end=('\n' if i == k-1 else ','))

main()

File: w2-list/cplusplus/TNL.cpp
#include <iostream>
//#include "LinkedList.cpp"
#include "ArrayList.cpp"
using namespace std;

int main()
{
    int k, b, t;
    cin >> k >> b;
    //LinkedList<int> bs;
    ArrayList<int> bs(b);
    for (int i = 0; i < b; i++)
    {
        int st;
        cin >> st;
        bs.append(st);
    }
    cin >> t;
    //LinkedList<int> ts;
    ArrayList<int> ts(t);
    for (int i = 0; i < t; i++)
    {
        int st;
        cin >> st;
        ts.append(st);
    }
    int task;
    cin >> task;
    switch (task)
    {
    case 1:
        for (int i = 0; i < k; i++)
        {
            cout << i;
            cout << (i == k - 1 ? "\n" : ",");
        }
        for (int i = 0; i < k; i++)
        {
            int index = bs.Search(i);
            if (index == -1)
            {
                cout << "," << (i == k - 1 ? "\n" : "");
            }
            else
            {
                cout << i << (i == k - 1 ? "\n" : ",");
            }
        }
        for (int i = 0; i < k; i++)
        {
            int index = ts.Search(i);
            if (index == -1)
            {
                cout << "," << (i == k - 1 ? "\n" : "");
            }
            else
            {
                cout << i << (i == k - 1 ? "\n" : ",");
            }
        }
    }
    return 0;
}